firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/TaskUtils.java

```diff
@@ -144,5 +145,12 @@ static <T> UpdateTask onSuccessUpdateTask(
     return updateTask;
   }
 
+  /** Set a {@link TaskCompletionSource} to be resolved with the result of another {@link Task}. */
+  static <T> void shadowTask(TaskCompletionSource<T> taskCompletionSource, Task<T> task) {
```

Comment:
Can/should this be a method on `TaskCompletionSource` itself?
If so, can/should it use the `TaskCompletionsSource`'s executor instead of the "direct" one?

Reply:
TaskCompletionSource is provided by the play-services library, so we can't add it there (although we could file a feature request).

Also just a note: `TaskCompletionSource` doesn't actually have its own executor. When you set a result on it, it passes that through to the underlying `TaskImpl`, which executes any handlers on the executors that were specified when they were added. Using direct executor here ensures that any handlers that were themselves added using a direct executor will behave as expected: they'll be executed on the thread that set the result. (I'll add a note to that effect).